mqtt/mqtt-broker/src/auth/authentication.rs

```diff
@@ -13,14 +15,21 @@ pub enum Credentials {
 #[derive(Clone, Debug)]
 pub struct Certificate(Vec<u8>);
 
+impl AsRef<[u8]> for Certificate {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_ref()
+    }
+}
+
 impl From<Vec<u8>> for Certificate {
     fn from(certificate: Vec<u8>) -> Self {
         Self(certificate)
     }
 }
 
 /// A trait to authenticate a MQTT client with given credentials.
-pub trait Authenticator {
+#[async_trait]
+pub trait Authenticator: Clone {
```

Comment:
I made this change, however I don't see the benefit to write the type spec at every place (3 at this moment) instead of bringing it with the trait via "inheritance"?